### PR TITLE
feat (optimize/3.8.0): redirect rules for optimize/3.8.0 docs

### PR DESCRIPTION
**Pull request by pepopowitz**
Child of #1234

My intention is to merge this into #1236 (not #1234), then #1236 into #1234, then #1234 into `main`. It's kind of complicated but this PR doesn't make sense without the content all having been moved.

## What is the purpose of the change

This PR adds redirects for all optimize/3.8.0 docs that have been moved. 

Included are a few pre-existing redirects against some Optimize docs, which have been updated to redirect to the correct optimize/3.8.0 location. 

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: static/.htaccess
Comment (pepopowitz) [inline static/.htaccess:308]: The only edit I made to these rules, all of which were moved from elsewhere in the file, is to update the "to" URL to be in the /optimize section instead of /docs/components/optimize.

File: static/.htaccess
Comment (pepopowitz) [inline static/.htaccess:310]: Most of the redirects in this file have two rules per URL. In a few cases it's to handle child routes, but in the large majority _I think_ it's to handle two cases each: when the URL contains a trailing slash, and when it doesn't. 

These new rules only contain one line per URL, but the `(\/?)` at the end of each catches both cases -- trailing slash or no trailing slash. 

If I'm missing another reason for the dual rules, let me know!

(Here's an example of another rule in the file, which uses two sub-rules): 

```
RewriteRule ^docs/self-managed/optimize-deployment/setup/configuration(.*)$ /docs/self-managed/optimize-deployment/configuration/system-configuration [R=301,L]
RewriteRule ^docs/self-managed/optimize-deployment/setup/configuration$ /docs/self-managed/optimize-deployment/configuration/system-configuration [R=301,L]
```

Comment (akeller) [reply static/.htaccess:310]: > when the URL contains a trailing slash, and when it doesn't.

Nailed it. Child routes and trailing slashes. 

We had to go back and forth a bit on trailing vs. no trailing slashes. I think we've committed to trailing slashes at this point so if you have a good way to reduce these to ONE rule, I'm all for it.

Comment (akeller) [reply static/.htaccess:310]: Plus I'm terrible at anything regex or regex leaning.

**Review by akeller — APPROVED** [review]
I see nothing obvious with these. 

As a bonus, it would be great to add the `(\/?)` into our docs on docs.

**Comment by christinaausley** [comment]
Thank you so much for executing these redirects. I'll plan to hold off on my "redirect clean up/pruning" until this is merged.

**Comment by pepopowitz** [comment]
Merging this into the parent branch, then I'll deal with any rebases that I need to.

**Pull request merged**
